src/graphing.module.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  SpelunkedEdge,
+  SpelunkedNode,
+  SpelunkedTree,
+} from './spelunker.interface';
+
+export class GraphingModule {
+  static graph(tree: SpelunkedTree[]): SpelunkedNode {
```

Comment:
The key point is to generate a more graph-like data structure like the proposed SpelunkedNode, which is agnostic of any specific graphing renderer, but makes it much easier to traverse the graph for whatever needs you might have.